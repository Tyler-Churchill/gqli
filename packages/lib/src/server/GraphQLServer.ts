import 'reflect-metadata';
import express from 'express';

import { ApolloServer } from 'apollo-server-express';
import { buildSchemaSync } from 'type-graphql';
import { GraphQLSchema } from 'graphql';

export class GraphQLServer {
  express: express.Application;
  server: ApolloServer;
  schema: GraphQLSchema;

  constructor(resolvers: Array<Function | string>) {
    // create an express app
    this.express = express();
    this.schema = buildSchemaSync({
      resolvers,
    });
    // create an apollo server with the newly created schema
    this.server = new ApolloServer({
      schema: this.schema,
      introspection: true,
      playground: true,
    });
    // inject express into apollos request middleware
    this.server.applyMiddleware({ app: this.express });
  }
  /**
   * Local development
   */
  async startLocal() {
    await this.express.listen({ port: 4000 }, () =>
      console.log(
        `ðŸš€ Server ready at http://localhost:4000${this.server.graphqlPath}`,
      ),
    );
  }
}
