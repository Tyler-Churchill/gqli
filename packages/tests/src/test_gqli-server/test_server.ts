import { createTestClient } from 'apollo-server-testing';
import { GraphQLServer } from '@gqli/server';
import gql from 'graphql-tag';

test('server boots', async () => {
  const server = new GraphQLServer();
  await server.start();
  const { query } = createTestClient(server.apollo);
  const VERSION_QUERY = gql`
    query {
      getVersion
    }
  `;
  const res = await query({
    query: VERSION_QUERY,
  });
  expect(res).toMatchSnapshot();
  await server.shutdown();
});
