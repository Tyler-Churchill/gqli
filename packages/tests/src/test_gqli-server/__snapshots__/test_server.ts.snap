// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server boots 1`] = `
Object {
  "data": Object {
    "getVersion": "1.0.0",
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
